certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -896,8 +898,50 @@ def test_preferred_chain(context: IntegrationTestsContext) -> None:
 
         dumped = misc.read_certificate(cert_path)
         assert 'Issuer: CN={}'.format(expected) in dumped, \
-               'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
+            'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
 
         with open(conf_path, 'r') as f:
             assert 'preferred_chain = {}'.format(requested) in f.read(), \
-                   'Expected preferred_chain to be set in renewal config'
+                'Expected preferred_chain to be set in renewal config'
+
+
+def test_certificate_validity(context: IntegrationTestsContext) -> None:
```

Comment:
There is one more thing I would like this test to do, and that is after the certificate issued, and also after it is renewed, to check that `certificate_validity` is present in the renewal parameters.

You can use `assert_saved_lineage_option` to do this.